src/main/java/io/jenkins/plugins/analysis/warnings/JUnit.java

```diff
@@ -0,0 +1,49 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import edu.hm.hafner.analysis.IssueParser;
+import edu.hm.hafner.analysis.parser.violations.JUnitAdapter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.jenkinsci.Symbol;
+import hudson.Extension;
+
+import io.jenkins.plugins.analysis.core.model.ReportScanningTool;
+
+/**
+ * Provides a parser and customized messages for JUnit.
+ *
+ * @author Gyanesha Prajjwal
+ */
+public class JUnit extends ReportScanningTool {
+    private static final long serialVersionUID = -5341616371387604827L;
+    private static final String ID = "junit";
+
+    /** Creates a new instance of {@link JUnit}. */
+    @DataBoundConstructor
+    public JUnit() {
+        super();
+        // empty constructor required for stapler
+    }
+
+    @Override
+    public IssueParser createParser() {
+        return new JUnitAdapter();
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Symbol("jUnit")
+    @Extension
+    public static class Descriptor extends ReportScanningToolDescriptor {
```

Comment:
Can you check if we can include the JUnit 5 logo as well? It would be niche to have the reports linked with the official new logo https://junit.org/junit5/.

Reply:
How will I link the reports with junit logo?
Should I download and store the logo into icons folder and use override getLabelProvider() in JUnit.java

Reply:
Yes, you need to overwrite that method, here is an example: https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/java/io/jenkins/plugins/analysis/warnings/EsLint.java#L69

Reply:
The icons belong to the icons folder and should be in 48x48 and 24x24 PNG. In the near feature I will change that to SVG but we can add that later on...

Reply:
Added logos of junit.

Reply:
Ok. thanks, Do you know if they did specify the license somewhere? Then we can add a corresponding line in https://github.com/jenkinsci/warnings-ng-plugin/blob/master/src/main/webapp/icons/LICENSE.txt

Reply:
They have a license for the source code distribution [https://github.com/junit-team/junit5/blob/master/LICENSE.md](https://github.com/junit-team/junit5/blob/master/LICENSE.md). 
`THE ACCOMPANYING PROGRAM IS PROVIDED UNDER THE TERMS OF THIS ECLIPSE PUBLIC LICENSE (“AGREEMENT”). ANY USE, REPRODUCTION OR DISTRIBUTION OF THE PROGRAM CONSTITUTES RECIPIENT'S ACCEPTANCE OF THIS AGREEMENT.`

Should I ask them ?

Reply:
That would be nice! I always have a bad feeling with icons that do not have the license correctly set.

Reply:
Will this work ? [https://github.com/junit-team/junit5/issues/2165](https://github.com/junit-team/junit5/issues/2165)

Reply:
Yes, wonderful! You can link to the GitHub issue in the license.txt file. 

Please also add the URL to JUnit with getUrl (see comment above) and then I will merge your changes!

Reply:
Sorry for not mentioning it beforehand, but I have already added the URL to JUnit with getUrl().